File: Java/Switcher/app/src/main/java/become/sk/javabutton/MainActivity.java
package become.sk.javabutton;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

/*
* Vylepši program tak, aby prepínal farbu pozadia.
* Zmeň meniaci sa text.
* Zmeň text v tlačidle.
*
* */

public class MainActivity extends AppCompatActivity {

    Boolean toggleTitle = false;
    TextView textView;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setTitle("Prepínač");

        button = (Button) findViewById(R.id.buttonX);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                prepni();
            }
        });

        textView = (TextView) findViewById(R.id.textViewX);
    }

    protected void prepni() {
        if (toggleTitle) {
            toggleTitle = false;
            textView.setText("Vyplo mi");
        } else {
            toggleTitle = true;
            textView.setText("Zaplo mi");
        }
    }
}

File: Kotlin/Switcher/app/src/main/java/become/sk/javabutton/MainActivity.kt
package become.sk.javabutton

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

/*
* Vylepši program tak, aby prepínal farbu pozadia.
* Zmeň meniaci sa text.
* Zmeň text v tlačidle.
*
* */

class MainActivity : AppCompatActivity() {

    internal var toggleTitle: Boolean? = false
    var textView: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        title = "Prepínač"

        val button = findViewById(R.id.buttonX) as Button
        button.setOnClickListener { prepni() }

        textView = findViewById(R.id.textViewX) as TextView
    }

    protected fun prepni() {
        if (toggleTitle!!) {
            toggleTitle = false
            textView!!.text = "Vyplo mi"
        } else {
            toggleTitle = true
            textView!!.text = "Zaplo mi"
        }
    }
}
